### ISSUE TITLE
Formatter::asPercent() should respect SI writing style rules

### ISSUE THREAD

**Issue by astwinioenergy** [yiisoft/yii2#19990]
### What steps will reproduce the problem?

echo Yii::$app->formatter->asPercent(10);

### What is the expected result?

1,000 %
(note extra space before % sign)

### What do you get instead?

1,000%

### Additional info

According to below rules: 
https://en.wikipedia.org/wiki/International_System_of_Units#SI_writing_style
almost all units should be written with extra space before a unit sign, including units like % and ℃.

**Comment by rob006** [comment]
AFAIK Yii uses intl for formatting in this case, and respects locale settings. And despite SI recommendations, general convention in English language is no space before `%`.

https://en.wikipedia.org/wiki/Percent_sign

**Comment by rob006** [comment]
> Yii does not use intl in case of %.

It does: https://github.com/yiisoft/yii2/blob/cfd8c6bfe98d39d320bf23a45778962fdcaebd84/framework/i18n/Formatter.php#L1223-L1230

**Comment by astwinioenergy** [comment]
> > Yii does not use intl in case of %.
> 
> It does:
> 
> https://github.com/yiisoft/yii2/blob/cfd8c6bfe98d39d320bf23a45778962fdcaebd84/framework/i18n/Formatter.php#L1223-L1230

This is true. My bad. Closing this issue as it's unrelated to yii. Thanks!
